pkg/trace/writer/trace.go

```diff
@@ -116,6 +116,8 @@ func NewTraceWriter(cfg *config.AgentConfig, prioritySampler samplerTPSReader, e
 	if s := cfg.TraceWriter.FlushPeriodSeconds; s != 0 {
 		tw.tick = time.Duration(s*1000) * time.Millisecond
 	}
+	qsize = 1
+	climit = 1
```

Comment:
This may be too low for climit. We need to test.

My local testing used a fake intake, so it was probably fast enough to keep up with whatever I sent to the agent.